src/Eventso.Subscription.Kafka.DeadLetter.Postgres/PoisonEventStore.cs

```diff
@@ -0,0 +1,386 @@
+            _connectionFactory = connectionFactory;
+            _maxAllowedFailureCount = maxAllowedFailureCount ?? 10;
+            _minIntervalBetweenRetries = minIntervalBetweenRetries ?? TimeSpan.FromMinutes(1);
+        }
+
+        public static async Task<PoisonEventStore> Initialize(
+            IConnectionFactory connectionFactory,
+            // TODO receive from settings
+            int? maxAllowedFailureCount = default,
+            TimeSpan? minIntervalBetweenRetries = default)
+        {
+            await using var connection = connectionFactory.ReadWrite();
+            
+            await using var command = new NpgsqlCommand(@"
+                CREATE SCHEMA IF NOT EXISTS eventso_dlq;
+
+                CREATE TABLE IF NOT EXISTS eventso_dlq.poison_events (
+                    topic                  TEXT         NOT NULL,
+                    partition              INT          NOT NULL,
+                    ""offset""               BIGINT       NOT NULL,
+                    key                    UUID         NOT NULL,
+                    value                  BYTEA        NULL,
+                    creation_timestamp     TIMESTAMP    NOT NULL,
+                    header_keys            TEXT[]       NULL,
+                    header_values          BYTEA[]      NULL,
+                    last_failure_timestamp TIMESTAMP    NOT NULL,
+                    last_failure_reason    TEXT         NOT NULL,
+                    total_failure_count    INT          NOT NULL,
+                    PRIMARY KEY (""offset"", partition, topic)
+                );
+
+                CREATE INDEX IF NOT EXISTS ix_poison_events_topic ON eventso_dlq.poison_events (topic);",
+                connection);
+
+            await connection.OpenAsync();
+            
+            await command.ExecuteNonQueryAsync();
+
+            return new PoisonEventStore(connectionFactory);
+        }
+
+        public async Task<long> Count(string topic, CancellationToken cancellationToken)
+        {
+            await using var connection = _connectionFactory.ReadOnly();
+
+            await using var command = new NpgsqlCommand(
+                "SELECT COUNT(*) FROM eventso_dlq.poison_events WHERE topic = @topic;",
+                connection)
+            {
+                Parameters =
+                {
+                    new NpgsqlParameter<string>("topic", topic)
+                }
+            };
+
+            await connection.OpenAsync(cancellationToken);
+
+            var result = await command.ExecuteScalarAsync(cancellationToken);
+            return result != null ? (long)result : 0;
+        }
+
+
+        public async Task<bool> IsStreamStored(string topic, Guid key, CancellationToken token)
+        {
+            await using var connection = _connectionFactory.ReadOnly();
+
+            await using var command = new NpgsqlCommand(
+                "SELECT TRUE FROM eventso_dlq.poison_events WHERE topic = @topic AND key = @key;",
```

Comment:
there is no index for this frequent query

Reply:
no it is (by key)